### ISSUE TITLE
Make light/dark mode experience more consistent

### ISSUE THREAD

**Issue by mcnesium** [hedgedoc/react-client#1235]
Referring to [this call](https://community.hedgedoc.org/t/help-wanted-ui-ux-designers/380) and the current state of [the UI demo](https://ui-test.hedgedoc.org/n/features) I would like to suggest to make the user experience more consistent regarding light vs. dark mode by making use of [`prefers-color-scheme`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme) by default.

Dark mode:

<img src="https://user-images.githubusercontent.com/476060/117059224-dd1a1e80-ad1f-11eb-8d26-40b1b7d46abc.png" width="294" height="163"/> <img src="https://user-images.githubusercontent.com/476060/117061524-9aa61100-ad22-11eb-8875-8772e7403589.png" width="30" height="163"/>  <img src="https://user-images.githubusercontent.com/476060/117059219-db505b00-ad1f-11eb-96a4-a3a9694b1b0f.png" width="294" height="163"/> 

Light mode:

<img src="https://user-images.githubusercontent.com/476060/117059222-dc818800-ad1f-11eb-839e-a18563576d4f.png" width="294" height="163"/> <img src="https://user-images.githubusercontent.com/476060/117061524-9aa61100-ad22-11eb-8875-8772e7403589.png" width="30" height="163"/>  <img src="https://user-images.githubusercontent.com/476060/117059860-8bbe5f00-ad20-11eb-9d7a-7bd088b717b5.png" width="294" height="163"/> 

Besides setting the modes manually using the :sun_with_face: and :crescent_moon: buttons for the mockups, notice the change of the status bar CSS for the light version:

    @media (prefers-color-scheme: light) {
      .status-bar {
        background: #eee;
        border-top: 1px solid #bcbcbc;
        color: #1c1c1c;
      }
    }

**Comment by ErikMichelson** [comment]
Hi @mcnesium,

thanks for your submission.

Changing the color of the status bar seems like a good idea.

I've got good news regarding the `prefers-color-scheme`: We already implemented it. The default theme (light or dark) is chosen based on your system preference. However, this setting is stored in your browser's local storage. So when revisiting the page the theme stored in local storage is used.

Anyway, I'd like to know your opinion on the following question:
 1. Currently the theme for the editor (left side) is toggled independently of the overall theme. Would it be more intuitive if it toggles with the overall theme?
 2. Do you prefer the system color scheme over your locally stored theme? So if you change the theme with the buttons, should that be saved only for the session or longer?

**Comment by mcnesium** [comment]
Great to see this being already in the pipeline…

>     1. Currently the theme for the editor (left side) is toggled independently of the overall theme. Would it be more intuitive if it toggles with the overall theme?

Yes, I think the appearance should be consistent throughout the entire application, even the entire screen. The obvious use case to swap between light and dark mode is when the surrounding light situation changes. When being in a bright light environment (e.g. daylight), a light user interface is easier to read, while in dark environments (e.g. at night) a dark interface is less distracting. 

This is true even for parts of the interface (anecdotal evidence: before Github introduced its dark mode, I used to just invert the screen to browse it at night, but then the black header bar turned white and sort of blinded). Thats why I suggested to even invert the status bar. Interface elements with a contrast to the base appearance should be those with an explicit functionality ("call to action").

Still, some people just like a dark editor better, while still prefering the rendered HTML to be "as everyone else will see it", so they should be free to switch the color to their liking.

>     2. Do you prefer the system color scheme over your locally stored theme? So if you change the theme with the buttons, should that be saved only for the session or longer?

As stated above, the prefered system color scheme should be the default, but since people might like something else better, their settings should not vanish, when they close the tab. So yes, once someone changed the defaults, this should be made persistent imho.

Thanks for asking :)
